# Exploring eBay Car Sales

Will be cleaning car listing data from eBay to allow analysis. Cleaning will be completed using pandas.

In [70]:
import pandas as pd
import numpy as np

To identify problematic areas, the dataset is imported and viewed.

In [71]:
autos = pd.read_csv("autos.csv", encoding = "Latin-1")

In [72]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offerType    price   abtest vehicleType  yearOfRegistration  \
0      privat   Angebot   $5,000  control         bus                2004   
1      privat   Angebot   $8,500  control   limousine                1997   
2      privat   Angebot   $8,990     test   limousine                2009   
3      privat   Angebot   $4,350  control  kleinwagen                2007   
4      privat   Angebot   $1,350     test       kombi                2003   
...       ...       ...      ...      ...         ...                 ...   
49995  privat   Angebot  $24,900  control   limousine                2011   
49996  privat   Angebot   $1,980  control      cabrio                1996   
49997  privat   Angebot  $13,200     test      cabrio                2014   
49998  privat   Angebot  $22,900  control       kombi                2013   
49999  privat   Angebot   $1,250  control   limousine                1996   

         gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0        manuell      158  andere  150,000km                    3      lpg   
1      automatik      286     7er  150,000km                    6   benzin   
2        manuell      102    golf   70,000km                    7   benzin   
3      automatik       71  fortwo   70,000km                    6   benzin   
4        manuell        0   focus  150,000km                    7   benzin   
...          ...      ...     ...        ...                  ...      ...   
49995  automatik      239      q5  100,000km                    1   diesel   
49996    manuell       75   astra  150,000km                    5   benzin   
49997  automatik       69     500    5,000km                   11   benzin   
49998    manuell      150      a3   40,000km                   11   diesel   
49999    manuell      101  vectra  150,000km                    1   benzin   

            brand notRepairedDamage          dateCreated  nrOfPictures  \
0         peugeot              nein  2016-03-26 00:00:00             0   
1             bmw              nein  2016-04-04 00:00:00             0   
2      volkswagen              nein  2016-03-26 00:00:00             0   
3           smart              nein  2016-03-12 00:00:00             0   
4            ford              nein  2016-04-01 00:00:00             0   
...           ...               ...                  ...           ...   
49995        audi              nein  2016-03-27 00:00:00             0   
49996        opel              nein  2016-03-28 00:00:00             0   
49997        fiat              nein  2016-04-02 00:00:00             0   
49998        audi              nein  2016-03-08 00:00:00             0   
49999        opel              nein  2016-03-13 00:00:00             0   

       postalCode             lastSeen  
0           79588  2016-04-06 06:45:54  
1           71034  2016-04-06 14:45:08  
2           35394  2016-04-06 20:15:37  
3           33729  2016-03-15 03:16:28  
4           39218  2016-04-01 14:38:50  
...           ...                  ...  
4999

In [73]:
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

- 20 columns in total
- All dates are in YYYY-MM-DD HH:mm:SS time as float
- 7 columns contain values that can be described as float if cleaned to remove strings and characters 
- Brands and other names may need to be cleaned so that different spellings or capitalizations are fixed
- some columns have null values
- spaces cannot just be replaced with underscores due to different text

In [74]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [75]:
autos.columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'powerps', 
       'model', 'odometer', 'registration_month', 'fueltype', 'brand',
       'unrepaired_damage', 'ad_created', 'nrofpictures', 'postal_code',
       'lastseen']

In [76]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  powerps   model   odometer  registration_month fueltype  \
0    manuell      158  andere  150,000km                   3      lpg   
1  automatik      286     7er  150,000km                   6   benzin   
2    manuell      102    golf   70,000km                   7   benzin   
3  automatik       71  fortwo   70,000km                   6   benzin   
4    manuell        0   focus  150,000km                   7   benzin   

        brand unrepaired_damage           ad_created  nrofpictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postal_code             lastseen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

## Cleaning

To begin cleaning:
- Columns with all or nearly all the same values are dropped
- Numeric data stored as text can be cleaned and converted

In [77]:
autos.describe()

registration_year       powerps  registration_month  nrofpictures  \
count       50000.000000  50000.000000        50000.000000       50000.0   
mean         2005.073280    116.355920            5.723360           0.0   
std           105.712813    209.216627            3.711984           0.0   
min          1000.000000      0.000000            0.000000           0.0   
25%          1999.000000     70.000000            3.000000           0.0   
50%          2003.000000    105.000000            6.000000           0.0   
75%          2008.000000    150.000000            9.000000           0.0   
max          9999.000000  17700.000000           12.000000           0.0   

        postal_code  
count  50000.000000  
mean   50813.627300  
std    25779.747957  
min     1067.000000  
25%    30451.000000  
50%    49577.000000  
75%    71540.000000  
max    99998.000000

Columns that have little to no use:
- nrofpictures has only zero
- seller either private or company
- registration month does not tell much about the car
- offer type has few values that are necessary

In [78]:
autos.drop(["nrofpictures","seller","registration_month","offer_type"],axis = 1)

date_crawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

         price   abtest vehicle_type  registration_year    gearbox  powerps  \
0       $5,000  control          bus               2004    manuell      158   
1       $8,500  control    limousine               1997  automatik      286   
2       $8,990     test    limousine               2009    manuell      102   
3       $4,350  control   kleinwagen               2007  automatik       71   
4       $1,350     test        kombi               2003    manuell        0   
...        ...      ...          ...                ...        ...      ...   
49995  $24,900  control    limousine               2011  automatik      239   
49996   $1,980  control       cabrio               1996    manuell       75   
49997  $13,200     test       cabrio               2014  automatik       69   
49998  $22,900  control        kombi               2013    manuell      150   
49999   $1,250  control    limousine               1996    manuell      101   

        model   odometer fueltype       brand unrepaired_damage  \
0      andere  150,000km      lpg     peugeot              nein   
1         7er  150,000km   benzin         bmw              nein   
2        golf   70,000km   benzin  volkswagen              nein   
3      fortwo   70,000km   benzin       smart              nein   
4       focus  150,000km   benzin        ford              nein   
...       ...        ...      ...         ...               ...   
49995      q5  100,000km   diesel        audi              nein   
49996   astra  150,000km   benzin        opel              nein   
49997     500    5,000km   benzin        fiat              nein   
49998      a3   40,000km   diesel        audi              nein   
49999  vectra  150,000km   benzin        opel              nein   

                ad_created  postal_code             lastseen  
0      2016-03-26 00:00:00        79588  2016-04-06 06:45:54  
1      2016-04-04 00:00:00        71034  2016-04-06 14:45:08  
2      2016-03-26 00:00:00        35394  2016-04-06 20:15:37  
3      2016-03-12 00:00:00        33729  2016-03-15 03:16:28  
4      2016-04-01 00:00:00        39218  2016-04-01 14:38:50  
...                    ...          ...                  ...  
49995  2016-03-27 00:00:00        82131  2016-04-01 13:47:40  
49996  2016-03-28 00:00:00        44807  2016-04-02 14:18:02  
49997  2016-04-02 00:00:00        73430  2016-04-04 11:47:27  
49998  2016-03-08 00:00:00        35683  2016-04-05 16:45:07  
49999  2016-03-13 00:00:00        45897  2016-04-06 21:18:48  

[50000 rows x 16 columns]

For numeric data that is text (odometer and price):
- remove non-numeric data
- convert column to numeric
- for odometer, rename to odometer_km

In [79]:
autos["price"] = autos["price"].str.replace("$","").str.replace(",","").astype(float)
autos["odometer"] = autos["odometer"].str.replace("km","").str.replace(",","").astype(float)
autos.rename({"odometer":"odometer_km"}, axis = 1, inplace = True)

In [80]:
print(autos["price"].unique().shape)
print("\n")
print(autos["price"].describe())
print("\n")
print(autos["price"].value_counts())
print("\n")
print(autos["price"].value_counts().sort_index(ascending=False).head(20))
print("\n")
print(autos["price"].value_counts().sort_index().head(10))

(2357,)


count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64


0.0        1421
500.0       781
1500.0      734
2500.0      643
1200.0      639
           ... 
6202.0        1
18310.0       1
898.0         1
11240.0       1
789.0         1
Name: price, Length: 2357, dtype: int64


99999999.0    1
27322222.0    1
12345678.0    3
11111111.0    2
10000000.0    1
3890000.0     1
1300000.0     1
1234566.0     1
999999.0      2
999990.0      1
350000.0      1
345000.0      1
299000.0      1
295000.0      1
265000.0      1
259000.0      1
250000.0      1
220000.0      1
198000.0      1
197000.0      1
Name: price, dtype: int64


0.0     1421
1.0      156
2.0        3
3.0        1
5.0        2
8.0        1
9.0        1
10.0       7
11.0       2
12.0       3
Name: price, dtype: int64


In [81]:
print(autos["odometer_km"].unique().shape)
print("\n")
print(autos["odometer_km"].describe())
print("\n")
print(autos["odometer_km"].value_counts())

(13,)


count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64


150000.0    32424
125000.0     5170
100000.0     2169
90000.0      1757
80000.0      1436
70000.0      1230
60000.0      1164
50000.0      1027
5000.0        967
40000.0       819
30000.0       789
20000.0       784
10000.0       264
Name: odometer_km, dtype: int64


Outliers to be removed:

For price, there are outliers that need to be removed above $20000 and below $200 as the prices above are single points and below are not values that cars normally are sold for.

In [82]:
autos = autos[autos["price"].between(200,20000)]

## Date Percentage

Caluculate percentage of each date_crawled, ad_created, and lastseen to rank by date.

In [83]:
print(autos[['date_crawled','ad_created','lastseen']][0:5])
print('\n')
print(autos['date_crawled'].str[:10])

          date_crawled           ad_created             lastseen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50


0        2016-03-26
1        2016-04-04
2        2016-03-26
3        2016-03-12
4        2016-04-01
            ...    
49993    2016-03-15
49994    2016-03-22
49996    2016-03-28
49997    2016-04-02
49999    2016-03-14
Name: date_crawled, Length: 45312, dtype: object


In [84]:
autos['date_crawled'].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending = True)

2016-03-05    0.025490
2016-03-06    0.014080
2016-03-07    0.036149
2016-03-08    0.033347
2016-03-09    0.033038
2016-03-10    0.032530
2016-03-11    0.032309
2016-03-12    0.037319
2016-03-13    0.015735
2016-03-14    0.036878
2016-03-15    0.034053
2016-03-16    0.029705
2016-03-17    0.031581
2016-03-18    0.012756
2016-03-19    0.034450
2016-03-20    0.037716
2016-03-21    0.037231
2016-03-22    0.032684
2016-03-23    0.032420
2016-03-24    0.029330
2016-03-25    0.031471
2016-03-26    0.032574
2016-03-27    0.030764
2016-03-28    0.034936
2016-03-29    0.033987
2016-03-30    0.034009
2016-03-31    0.031625
2016-04-01    0.033413
2016-04-02    0.035487
2016-04-03    0.038665
2016-04-04    0.036679
2016-04-05    0.013065
2016-04-06    0.003178
2016-04-07    0.001346
Name: date_crawled, dtype: float64

Most data was scraped between March 5, 2016 and April 7, 2016.

In [85]:
autos['ad_created'].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending = True)

2015-08-10    0.000022
2015-09-09    0.000022
2015-11-10    0.000022
2015-12-05    0.000022
2015-12-30    0.000022
                ...   
2016-04-03    0.038908
2016-04-04    0.037032
2016-04-05    0.011763
2016-04-06    0.003244
2016-04-07    0.001214
Name: ad_created, Length: 74, dtype: float64

Ads were created from August 10, 2015 to April 7, 2016. Higher percentage of ads were created during the web scraping period and just prior to it.

In [86]:
autos['lastseen'].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending = True)

2016-03-05    0.001148
2016-03-06    0.004480
2016-03-07    0.005495
2016-03-08    0.007481
2016-03-09    0.009953
2016-03-10    0.010748
2016-03-11    0.012734
2016-03-12    0.024718
2016-03-13    0.009181
2016-03-14    0.012602
2016-03-15    0.015934
2016-03-16    0.016640
2016-03-17    0.028712
2016-03-18    0.007371
2016-03-19    0.016022
2016-03-20    0.021054
2016-03-21    0.020811
2016-03-22    0.021871
2016-03-23    0.018935
2016-03-24    0.019951
2016-03-25    0.019642
2016-03-26    0.016927
2016-03-27    0.015779
2016-03-28    0.021275
2016-03-29    0.022908
2016-03-30    0.024938
2016-03-31    0.024254
2016-04-01    0.023106
2016-04-02    0.025115
2016-04-03    0.025269
2016-04-04    0.025137
2016-04-05    0.122859
2016-04-06    0.217956
2016-04-07    0.128995
Name: lastseen, dtype: float64

All ads were last seen sometime within the scraping period with the most occurring at the end of the scraping period.

In [87]:
autos["registration_year"].describe()

count    45312.000000
mean      2004.332428
std         82.140145
min       1000.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

The average registration was from 2004 though outliers exist of 1000 and 9999.

In [88]:
autos[autos["registration_year"].between(1950,2016)].describe()

price  registration_year       powerps    odometer_km  \
count  43459.000000       43459.000000  43459.000000   43459.000000   
mean    4663.510320        2002.691042    113.458685  128664.028164   
std     4526.427616           6.694986    187.136498   36918.652086   
min      200.000000        1950.000000      0.000000    5000.000000   
25%     1250.000000        1999.000000     75.000000  125000.000000   
50%     2990.000000        2003.000000    105.000000  150000.000000   
75%     6750.000000        2007.000000    143.000000  150000.000000   
max    20000.000000        2016.000000  17700.000000  150000.000000   

       registration_month  nrofpictures   postal_code  
count        43459.000000       43459.0  43459.000000  
mean             5.854207           0.0  50834.336340  
std              3.664845           0.0  25717.286083  
min              0.000000           0.0   1067.000000  
25%              3.000000           0.0  30457.000000  
50%              6.000000           0.0  49610.000000  
75%              9.000000           0.0  71364.000000  
max             12.000000           0.0  99998.000000

Range of car registration year is limited between 1950 and 2016 since there is unlikely cars registered before 1950 and no cars are able to be registered after 2016.

In [89]:
autos = autos[autos["registration_year"].between(1950,2016)]

In [90]:
autos['registration_year'].value_counts(normalize=True)

2000    0.069284
2005    0.066338
1999    0.065303
2003    0.061621
2004    0.061437
          ...   
1958    0.000069
1954    0.000046
1953    0.000023
1950    0.000023
1952    0.000023
Name: registration_year, Length: 64, dtype: float64

Most cars were registered in the early 2000s and late 1990s.

In [91]:
autos["brand"].value_counts(normalize=True, dropna=False)

volkswagen        0.214248
opel              0.110817
bmw               0.107297
mercedes_benz     0.092225
audi              0.081410
ford              0.070894
renault           0.048828
peugeot           0.031501
fiat              0.026416
seat              0.018822
skoda             0.016843
nissan            0.015831
mazda             0.015647
smart             0.015141
citroen           0.014727
toyota            0.013438
hyundai           0.010562
volvo             0.009457
mini              0.008675
mitsubishi        0.008491
sonstige_autos    0.008376
honda             0.008284
kia               0.007248
alfa_romeo        0.006949
suzuki            0.006098
chevrolet         0.005753
chrysler          0.003751
dacia             0.002830
daihatsu          0.002623
subaru            0.002163
porsche           0.002048
jeep              0.002048
saab              0.001749
daewoo            0.001565
land_rover        0.001542
rover             0.001381
trabant           0.001358
j

In interest of the main car companies found in the data, the companies that make up 1% or more will be included in the analysis. 1% of the current data set is 43 cars and fewer than that is a reduction in statistical significance.

In [117]:
mean_price = {}
for b in autos["brand"].value_counts(normalize=True, dropna=False).index[:17]:
    mean_price[b] = sum(autos.loc[autos["brand"]==b,"price"])/(autos[autos["brand"]==b].shape[0])

In [118]:
mean_price

{'volkswagen': 4676.346901514338,
 'opel': 2942.3486295681064,
 'bmw': 6389.2459789834875,
 'mercedes_benz': 6127.65743512974,
 'audi': 6573.878745053703,
 'ford': 3330.3855890944496,
 'renault': 2443.0589066918,
 'peugeot': 3111.1869978086193,
 'fiat': 2906.726480836237,
 'seat': 4182.60880195599,
 'skoda': 5861.924863387978,
 'nissan': 4456.514534883721,
 'mazda': 3746.4573529411764,
 'smart': 3596.40273556231,
 'citroen': 3603.603125,
 'toyota': 4860.592465753424,
 'hyundai': 5255.337690631808}

The most expensive cars on average are BMW, Mercedes_benz, and Audi. Renault and opel are the least expensive.

In [121]:
mean_mileage = {}
for b in autos["brand"].value_counts(normalize=True, dropna=False).index[:17]:
    mean_mileage[b] = sum(autos.loc[autos["brand"]==b,"odometer_km"])/(autos[autos["brand"]==b].shape[0])

In [123]:
mean_mileage

{'volkswagen': 131537.4288475996,
 'opel': 129852.57475083056,
 'bmw': 137455.50075058974,
 'mercedes_benz': 136805.1397205589,
 'audi': 137355.850763143,
 'ford': 125770.85361895489,
 'renault': 128409.51932139491,
 'peugeot': 127063.55003652301,
 'fiat': 117033.97212543554,
 'seat': 123062.34718826406,
 'skoda': 113510.92896174864,
 'nissan': 119484.01162790698,
 'mazda': 125992.64705882352,
 'smart': 99734.0425531915,
 'citroen': 120484.375,
 'toyota': 117260.27397260274,
 'hyundai': 107461.87363834423}

In [126]:
bmp = pd.Series(mean_price)
bmm = pd.Series(mean_mileage)

In [133]:
brand_pm = pd.DataFrame(bmp, columns = ['mean_price'])
brand_pm["mean_mileage"] = bmm
brand_pm

mean_price   mean_mileage
volkswagen     4676.346902  131537.428848
opel           2942.348630  129852.574751
bmw            6389.245979  137455.500751
mercedes_benz  6127.657435  136805.139721
audi           6573.878745  137355.850763
ford           3330.385589  125770.853619
renault        2443.058907  128409.519321
peugeot        3111.186998  127063.550037
fiat           2906.726481  117033.972125
seat           4182.608802  123062.347188
skoda          5861.924863  113510.928962
nissan         4456.514535  119484.011628
mazda          3746.457353  125992.647059
smart          3596.402736   99734.042553
citroen        3603.603125  120484.375000
toyota         4860.592466  117260.273973
hyundai        5255.337691  107461.873638

Audi has the highest number of kilometers on average and Smart has the fewest. Price does not seem to correlate with mileage.